src/app/common/hooks/use-send-form-validation.ts

```diff
@@ -66,10 +67,12 @@ export const useStacksSendFormValidation = ({
       stxAmountSchema(formatPrecisionError('STX', STX_DECIMALS)).test({
         message: formatInsufficientBalanceError(stacksBalances?.stx.availableStx.amount, 'STX'),
         test(value: unknown) {
-          const fee = stxToMicroStx(this.parent.fee);
+          const fee = unitToFractionalUnit(STX_DECIMALS)(this.parent.fee);
```

Comment:
Sorry for the back and forth. Creating a factory function is a way to share logic that was previously repeated between stx/btc and other currencies, encapsulating the implementation detail of shifting a number by _n_.

That doesn't mean we should call the factory directly every time we want to use it. Now need to import both `unitToFractionalUnit` and `STX_DECIMALS`. This leaks the detail of the decimals, that it didn't before, and also impacts readability (stxToMicroStx is clearer than `unitToFractionalUnit(STX_DECIMALS)`.

```ts
function fractionalUnitToUnit(decimals: number) { .. }
export const satToBtc = fractionalUnitToUnit(BTC_DECIMALS);
export const microStxToStx = fractionalUnitToUnit(STX_DECIMALS); // 👈🏼  this API is exactly the same as before so doesn't require any changes
```

The resulting API that's consumed doesn't need to change at all if we export and consume the resulting functions.

Reply:
Got it, misread original comment. Will change.